src/vs/workbench/parts/terminal/common/terminal.ts

```diff
@@ -410,10 +410,11 @@ export interface ITerminalInstance {
 	/**
 	 * Dispose the terminal instance, removing it from the panel/service and freeing up resources.
 	 *
-	 * @param isShuttingDown Whether VS Code is shutting down, if so kill any terminal processes
-	 * immediately.
+	 * @param immediate Whether the kill should be immediate or not. Immedeate should only be used
+	 * when VS Code is shutting down or in cases where the terminal dispose was user initiated.
```

Comment:
Might be good to be more explicit about why `!immediate` exists here (to cover an edge case where final output can get cut off).